src/registry.rs

```diff
@@ -58,7 +58,7 @@ impl RegistryCore {
             // the collector_id.
             if desc_id_set.insert(desc.id) {
                 // The set did not have this value present, true is returned.
-                collector_id += desc.id;
+                collector_id ^= desc.id;
```

Comment:
Why does `collector_id` need to plus `desc.id` in the past? Why not just plus 1?